_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+One template for reporting issues is through a "user story".
+A user story typically comes in the form: `As a < type of user >, I want < some goal > so that < some reason >.` ([Mountain Goat Software: User Stories](https://www.mountaingoatsoftware.com/agile/user-stories)).
+Alongside the story, it can help to add in a snippet of code which exemplifies a problem, new functionality, or a potential adjacent / similar solution.
+As a general principle, be as __be as specific as you can without going overboard__.
+Include things like programming language version, operating system, and other system dependencies that might be related.
+Once you have a good written description of the need be sure to submit it where it can be seen by the relevant development community.
+For GitHub-based work, this is usually a GitHub Issue, but can also entail discussion board posts to gather buy-in or consensus before proceeding.
+
+### What Happens After You Submit a Bug or Feature Report?
+
+When making open-source contributions, sometimes it can also help to mention that you're interested in resolving the issue through a related pull request and review.
+Oftentimes open-source projects welcome new contributors but may have specific requirements.
+These requirements are usually spelled out within a [`CONTRIBUTING.md` document](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) found somewhere in the repository or the organization level documentation.
+It's also completely okay to let other contributors build solutions for the issue (like we mentioned before, all contributions matter, including the reporting of bugs or features themselves)!
+
+## Developing and Testing an Open-source Contribution
+
+### Creating a Development Workspace
+
+Once ready to develop a solution for the reported need in the open-source project you'll need a place to version your updates.
+This work generally takes place through version control on focused branches which are named in a way that relates to the focus.
+When working on GitHub, this work also commonly takes place on [forked repository copies](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+Using these methods helps isolate your changes from other work that takes place within the project.
+It also can help you track your progress alongside related changes that might take place before you're able to seek review or code merges.
+
+#### Bug or Feature Verification with Test-driven Development
+
+> 1. Add or modify a test which checks for a bug fix or feature addition
+> 2. Run all tests (expecting the newly added test content to fail)
+> 3. Write a simple version of code which allows the tests to succeed
+> 4. Verify that all tests now pass
+> 5. Return to step 3, refactoring the code as needed
+
```

Comment:
Are these steps supposed to be below the line further down that says "Test-driven development entails the following pattern:"?

Reply:
Whoops, yes, thanks! I tried to make these steps a sort of "graphic" ahead of the text, but they lost the improved flow you mention. I'll work this in somehow with an additional commit for this PR.